lib/Helper/Translate.php

```diff
@@ -120,6 +120,55 @@ public static function InitLocale($config, $language = NULL)
         self::$jsLocaleRequested = str_replace('_', '-', self::$requestedLanguage);
     }
 
+    /**
+     * Get translations for user selected language
+     * @param $config
+     * @param $language
+     * @return Translator
+     */
+    public static function getTranslationsFromLocale($config, $language = NULL): Translator
+    {
+        // The default language
+        $default = ($language === null) ? $config->getSetting('DEFAULT_LANGUAGE') : $language;
+
+        // Build an array of supported languages
+        $localeDir = PROJECT_ROOT . '/locale';
+        $supportedLanguages = array_map('basename', glob($localeDir . '/*.mo'));
+
+        // Record any matching languages we find.
+        $foundLanguage = null;
+
+        // Try to get the local firstly from _REQUEST (post then get)
+        if ($language != null) {
+            // Serve only the requested language
+            // Firstly, Sanitize it
+            self::$requestedLanguage = str_replace('-', '_', $language);
```

Comment:
We need to avoid changing any of the member variables, because they would also be used by the main CMS